pkg/metrics/composite_tags.go

```diff
@@ -0,0 +1,115 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022-present Datadog, Inc.
+
+package metrics
+
+import (
+	"encoding/json"
+	"strings"
+)
+
+// CompositeTags stores two tag sets and provides methods to manipulate them easily
+type CompositeTags struct {
+	tags1 []string
+	tags2 []string
+}
+
+// NewCompositeTags creates a new CompositeTags
```

Comment:
Worth mentioning that it's OK to pass `nil` for the second argument.

In fact, it might be more ergonomic if this was private and the public constructors were
 * `NewCompositeTagsFromSlice([]string) *CompositeTags`
 * `NewCompositeTagsCombine(*CompositeTags, []string) *CompositeTags`

Reply:
`NewCompositeTags` should not be private. I have updated `newContext` to use it because `newContext` is going to take two `tags.Entry` soon.

>  Worth mentioning that it's OK to pass nil for the second argument

`nil` is always a valid value for `[]string`. What would be the reason here to mention that it's OK to pass `nil`?